src/MaterialEditor.Core.Studio/Core.MaterialEditor.TimelineCompatibilityHelper.cs

```diff
@@ -534,5 +538,59 @@ internal static void RefreshInterpolablesList()
             }
             refreshInterpolablesList.Invoke(null, null);
         }
+
+        public static IEnumerable<int> GetUsedTextureIds()
+        {
+            var ids = new List<int>();
+            if (!IsTimelineAvailable()) return ids;
+
+            var interpolables = (IEnumerable)_GetAllInterpolables.Invoke(null, new object[] { false });
+
+            foreach (var interpolable in interpolables)
+            {
+                if (
+                    (string)_interpolableOwner.GetValue(interpolable) == "MaterialEditor"
+                    && (string)_interpolableId.GetValue(interpolable) == "textureProperty"
+                )
+                {
+                    var keyframes = (IDictionary)_interpolableKeyframes.GetValue(interpolable);
+                    foreach (var keyframe in keyframes.Values)
+                    {
+                        int id = (int)_keyframeValue.GetValue(keyframe);
+                        if (!ids.Contains(id)) ids.Add(id);
+                    }
+                }
+            }
+            return ids;
+        }
+
+        private static Type _timelineType;
+        private static Type _interpolableType;
+        private static Type _keyframeType;
+
+        private static MethodInfo _GetAllInterpolables;
+
+        private static FieldInfo _interpolableOwner;
+        private static FieldInfo _interpolableId;
+        private static FieldInfo _interpolableOci;
+        private static FieldInfo _interpolableKeyframes;
+        private static FieldInfo _keyframeValue;
+
+        private static void InitTimelineTypes()
+        {
+            _timelineType = System.Type.GetType("Timeline.Timeline,Timeline", throwOnError: false);
+            if (_timelineType == null) return;
+
+            _interpolableType = System.Type.GetType("Timeline.Interpolable,Timeline", throwOnError: false);
+            _keyframeType = System.Type.GetType("Timeline.Keyframe,Timeline", throwOnError: false);
+
+            _GetAllInterpolables = _timelineType.GetMethod("GetAllInterpolables", AccessTools.all);
+
+            _interpolableOwner = _interpolableType.GetField("owner", AccessTools.all);
+            _interpolableId = _interpolableType.GetField("id", AccessTools.all);
+            _interpolableOci = _interpolableType.GetField("oci", AccessTools.all);
+            _interpolableKeyframes = _interpolableType.GetField("keyframes", AccessTools.all);
+            _keyframeValue = _keyframeType.GetField("value", AccessTools.all);
```

Comment:
Add some sort of sanity checks in case any of these fields get changed in timeline.